File: src/main/java/tfg/proyecto/TFG/dtos/DTOtarjetaBancariaBajada.java
package tfg.proyecto.TFG.dtos;

import java.math.BigDecimal;
import java.time.LocalDate;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@Builder
@Data
@NoArgsConstructor
@AllArgsConstructor
public class DTOtarjetaBancariaBajada {
    private Long id;
    private String nombreTitular;
    private String nTarjeta;
    private LocalDate fechaCaducidad;
    private String cvv;  
    private BigDecimal saldo;

}

File: appMovilShowPass/app/src/main/java/com/example/appmovilshowpass/data/remote/dto/DTOtarjetaBancariaBajada.kt
package com.example.appmovilshowpass.data.remote.dto

import com.example.appmovilshowpass.model.TarjetaBancaria
import com.google.gson.annotations.SerializedName
import java.math.BigDecimal
import java.time.LocalDate

data class DTOtarjetaBancariaBajada(
    val id: Long = 0,
    val nombreTitular: String = "",
    val fechaCaducidad: String = "",
    val cvv: String = "",
    val saldo: BigDecimal = BigDecimal.ZERO,
    @SerializedName("ntarjeta")
    val nTarjeta: String = "",
)

fun DTOtarjetaBancariaBajada.toTarjetaBancaria(): TarjetaBancaria {
    return TarjetaBancaria(
        id = this.id,
        nombreTitular = this.nombreTitular ?: "",
        fechaCaducidad = if (this.fechaCaducidad.isNotBlank()) LocalDate.parse(this.fechaCaducidad) else LocalDate.now(),
        cvv = this.cvv ?: "",
        saldo = this.saldo ?: BigDecimal.ZERO,
        nTarjeta = this.nTarjeta ?: ""
    )

}